pkg/secrets/host/store.go

```diff
@@ -0,0 +1,20 @@
+package host
+
+import (
+	"get.porter.sh/porter/pkg/secrets"
+	"github.com/cnabio/cnab-go/credentials"
+	"github.com/cnabio/cnab-go/credentials/host"
+)
+
+var _ secrets.Store = &Store{}
+
+// Store implements a store that resolves secrets from the host.
+type Store struct {}
+
+func NewStore() *Store {
+	return &Store{}
+}
+
+func (s *Store) Resolve(cred credentials.Source) (string, error) {
+	return host.Resolve(cred)
```

Comment:
Confused on what `host` designates.  Different from filesystem? Env vars?  Other?

The `credentials/host` pkg must be on a branch of `cnab-go` @carolynvs ?  Perhaps when that PR can be issued, context will help understanding.

Reply:
Yes, this entire PR relies upon a branch of cnab-go that I'm terrified of splitting up and trying to get merged upstream. 😬 

https://github.com/carolynvs/cnab-go/tree/credentials

credentials/host is a package that takes the logic for resolving credentialsets against the host environment out of credential sets. I am trying to make credential sets generic so that their values can be resolved against a variety of sources, for example azure key vault, a company's hand rolled secret system, hashicorp's vault, instead of assuming that they can only come from the host and only a hard coded set of locations (env, command, path or value).

Reply:
Thank you, I see now.  Perhaps those changes can be PRed upstream if https://github.com/carolynvs/cnab-go/blob/credentials/credentials/credentialset.go is thinned out to perhaps just structs/interfaces and its pre-existing default logic to resolve via the host/filesystem can now use code in the `host` pkg you've added?  (I'm probably missing some details here :) )

Reply:
Yeah starting Monday I'll strip that branch down for parts and start submitting PRs and see what we can trick people into accepting. 😉 

I'm anticipating the problems around changing the crud.Store interface to accept an itemType. That's going to be a breaking change if anyone is using it directly instead of using ClaimStore. So we will see! 🤞

https://github.com/carolynvs/cnab-go/blob/60b3c613dd4f92c8b79ead33d33ecfa11f3b2581/utils/crud/store.go#L4-L9